WordPress/WordPressTest/Dashboard/Dynamic Cards/BlogDashboardDynamicCardCoordinatorTests.swift

```diff
@@ -0,0 +1,87 @@
+import Nimble
+@testable import WordPress
+import XCTest
+
+final class BlogDashboardDynamicCardCoordinatorTests: XCTestCase {
+
+    override func setUp() {
+        super.setUp()
+
+        // Because AnalyticsEventTrackingSpy logs events in a static var, we need to reset it between tests
+        AnalyticsEventTrackingSpy.reset()
+    }
+
+    func test_trackAnalyticEventWhenDidAppearIsCalled() {
+        // Given
+        let id = "123"
+        let event = DashboardDynamicCardAnalyticsEvent.cardShown(id: id)
+        let coordinator = makeCoordinator(id: id)
+
+        // When
+        coordinator.didAppear()
+        coordinator.didAppear()
+        coordinator.didAppear()
+
+        // Then
+        expect(AnalyticsEventTrackingSpy.trackedEvents).to(haveCount(1))
+        expect(AnalyticsEventTrackingSpy.trackedEvents).to(containElementSatisfying({ firedEvent in
+            firedEvent.name == event.name && firedEvent.properties == event.properties
+        }))
+    }
+
+    func test_trackAnalyticEventWhenDidTapCardIsCalled() {
```

Comment:
We can probably have another test case for the `once` scenario where the user tap is invoked multiple times but we expect only a single event if I understood it correctly. (Also for the test below)

Reply:
Right now tap events are fired every time the user taps the card or the CTA, and I think this is the behavior we want, no?

Whereas for the card shown event, it is fired only once because the dequeue mechanism of collection view cells can cause the event to be fired multiple times when scrolling even though the user has seen the card only once.

Let me known that makes sense.

Reply:
I see. I thought it was the reason we trigger 3 events back to back. I misjudged it then.